pkg/ovs/openflow/ofctrl_bridge_test.go

```diff
@@ -160,3 +166,70 @@ func TestOFBridgePacketRcvd(t *testing.T) {
 		}
 	}
 }
+
+func TestOFMeterStats(t *testing.T) {
+	// Reset OVS metrics (Prometheus) and reinitialize them to test.
+	legacyregistry.Reset()
+	metrics.InitializeOVSMetrics()
+
+	b := NewOFBridge("test-br", GetMgmtAddress(ovsconfig.DefaultOVSRunDir, "test-br"))
+	sw := newFakeOFSwitch(b)
+	b.SetOFSwitch(sw)
+
+	mpMeterStatsReply := openflow15.NewMpReply(openflow15.MultipartType_MeterStats)
+	newMeterStats := func(meterID uint32, packets uint64) *openflow15.MeterStats {
+		meterStats := openflow15.NewMeterStats(meterID)
+		bandStats := openflow15.NewMeterBandStats()
+		bandStats.PacketBandCount = packets
+		meterStats.AddBandStats(*bandStats)
+		return meterStats
+	}
+	meterStats1 := newMeterStats(1, 100)
+	meterStats2 := newMeterStats(2, 0)
+	// Unexpected case
+	meterStats3 := newMeterStats(3, 300)
+	mpMeterStatsReply.Body = append(mpMeterStatsReply.Body, meterStats1)
+	mpMeterStatsReply.Body = append(mpMeterStatsReply.Body, meterStats2)
+	mpMeterStatsReply.Body = append(mpMeterStatsReply.Body, meterStats3)
+	expectedOVSMeterPacketDroppedCount := `
+    # HELP antrea_agent_ovs_meter_packet_dropped_count [ALPHA] Number of packets dropped by OVS meter. The value is greater than 0 when the packets exceed the rate-limit.
+    # TYPE antrea_agent_ovs_meter_packet_dropped_count gauge
+    antrea_agent_ovs_meter_packet_dropped_count{meter_id="PacketInMeterNetworkPolicy"} 100
+    antrea_agent_ovs_meter_packet_dropped_count{meter_id="PacketInMeterTraceflow"} 0
+	`
+	handleMeterStatsReply := func(packetCounts map[int]int64) {
+		for k, v := range packetCounts {
+			switch k {
+			case 1:
+				metrics.OVSMeterPacketDroppedCount.WithLabelValues("PacketInMeterNetworkPolicy").Set(float64(v))
+			case 2:
+				metrics.OVSMeterPacketDroppedCount.WithLabelValues("PacketInMeterTraceflow").Set(float64(v))
```

Comment:
I don't understand why we are using Prometheus metrics at all in this unit test. This `handleMeterStatsReply` callback function does not need to use metrics. This file is testing the `pkg/ovs/openflow` package, which has nothing to do with metrics. Instead you could use a map protected by a mutex, or a channel.

Reply:
I removed the part of checking Prometheus metrics and only checked the packet count in the latest change. Could you help take a look at the latest version?